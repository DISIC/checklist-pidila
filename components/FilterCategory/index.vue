<template>
  <fieldset class="filter-type fieldset-discrete">
    <legend>Filtrer par {{title}}</legend>
    <p class="filter-all">
      <input
        v-model="allIsChecked"
        class="filter_input checkbox"
        value="Tous"
        :id="formattedId('all')"
        type="checkbox"/>
      <label :for="formattedId('all')">Tous</label>
    </p>
    <ul class="list-unstyled">
      <li v-for="tag in data.tags" class="filter">
        <input
          v-model="checkedTags"
          class="filter_input checkbox"
          :value="tag"
          :id="formattedId(tag)"
          type="checkbox"/>
        <label :for="formattedId(tag)">{{tag}}</label>
      </li>
    </ul>
    <p class="filter-link">
      <anchor-link :anchor="'criteria-list'" :css-class="'sr-only sr-only-focusable'" :message="'Accéder à la liste des critères'"/>
    </p>
  </fieldset>
</template>

<script>
  import AnchorLink from '../AnchorLink/index'

  export default {
    components: {AnchorLink},
    name: 'FilterCategory',
    props: {
      data: {
        type: Object,
        default: () => {
          return {
            title: 'Filter is missing',
            tags: []
          }
        }
      }
    },
    methods: {
      formattedId: function (tag) {
        return this.data.title + '_' + tag.replace(/ /g, '_')
      }
    },
    computed: {
      title: function () {
        return this.data.title.toLowerCase()
      },
      checkedTags: {
        get: function () {
          return this.$store.getters.tagsFromFilterTitle(this.data.title)
        },
        set: function (value) {
          const payload = {
            title: this.data.title,
            tags: value
          }
          this.$store.commit('UPDATE_FILTER_TAGS', payload)
        }
      },
      allIsChecked: {
        get: function () {
          return this.checkedTags.length === 0
        },
        set: function () {
          this.checkedTags = []
        }
      }
    }
  }
</script>
