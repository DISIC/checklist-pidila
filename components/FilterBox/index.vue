<template>
  <div class="filter-boxes">
    <h2>
      <svg class="svg-icon icon-filter" aria-hidden="true" focusable="false">
        <use xlink:href="svg/icon-sprite.svg#filter"></use>
      </svg>
      Filtrer la liste
    </h2>

    <form id="filters" class="filters nav-box-content">
      <p class="filter-boxes-info">La sélection de filtres met à jour la liste des critères sans rechargement de page.</p>

      <filter-category v-for="filter in data" :key="filter.title" :data="filter"/>

    </form>
  </div>
</template>

<script>
  import AnchorLink from '../AnchorLink/index'

  import FilterCategory from '../FilterCategory/index'

  export default {
    components: {AnchorLink, FilterCategory},
    name: 'FilterBox',
    props: {
      data: {
        type: Array,
        default: () => {
          return [
            {title: 'Filter is missing', tags: []}
          ]
        }
      }
    }
  }
</script>
