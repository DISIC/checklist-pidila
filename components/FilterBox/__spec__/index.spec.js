import {shallowMount} from '@vue/test-utils'
import FilterCategory from '../../FilterCategory/index.vue'
import FilterBox from '../index.vue'

describe('components | FilterBox', () => {

  let props
  beforeEach(() => {
    props = {
      data: [
        {title: '0', tags: ['0-1', '0-2']},
        {title: '1', tags: ['1-1', '1-2']},
        {title: '2', tags: ['2-1', '2-2']},
      ]
    }
  })

  it('should match snapshot', () => {
    // when
    const wrapper = shallowMount(FilterBox)

    // then
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain three FilterCategory components when store has three filters', () => {
    // when
    const wrapper = shallowMount(FilterBox, {propsData: props})

    // then
    const filterCategoryComponents = wrapper.findAll(FilterCategory)
    expect(filterCategoryComponents).toHaveLength(3)
  })

  it('should set the right props in the FilterCategory components', () => {
    // when
    const wrapper = shallowMount(FilterBox, {propsData: props})

    // then
    const filterCategoryComponents = wrapper.findAll(FilterCategory)
    filterCategoryComponents.wrappers.forEach((component, index) => {
      expect(component.props()).toEqual({
        data: {
          title: String(index),
          tags: [String(index) + '-1', String(index) + '-2']
        }
      })
    })
  })
})
