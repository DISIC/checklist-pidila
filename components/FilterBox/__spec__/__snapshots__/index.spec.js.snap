// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | FilterBox should match snapshot 1`] = `
<div
  class="filter-boxes"
>
  <h2>
    <svg
      aria-hidden="true"
      class="svg-icon icon-filter"
      focusable="false"
    >
      <use
        xlink:href="svg/icon-sprite.svg#filter"
      />
    </svg>
    
    Filtrer la liste
  
  </h2>
   
  <form
    class="filters nav-box-content"
    id="filters"
  >
    <p
      class="filter-boxes-info"
    >
      La sélection de filtres met à jour la liste des critères sans rechargement de page.
    </p>
     
    <filter-category-stub
      data="[object Object]"
    />
  </form>
</div>
`;
